src/pl/senla/hotel/ie/ReaderWriter.java

```diff
@@ -0,0 +1,128 @@
+package pl.senla.hotel.ie;
+
+import com.opencsv.*;
+import pl.senla.hotel.entity.Guest;
+import pl.senla.hotel.entity.Order;
+import pl.senla.hotel.entity.facilities.HotelFacility;
+import pl.senla.hotel.entity.facilities.Room;
+import pl.senla.hotel.entity.services.HotelService;
+import pl.senla.hotel.entity.services.RoomReservation;
+import pl.senla.hotel.utils.GuestUtil;
+import pl.senla.hotel.utils.OrderUtil;
+import pl.senla.hotel.utils.RoomReservationUtil;
+import pl.senla.hotel.utils.RoomUtil;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static pl.senla.hotel.constant.InputOutputConstant.*;
+
+public class ReaderWriter<T> implements ReaderWriterUniversal<T>{
+
+    private final Header header;
+
+    public ReaderWriter() {
+        this.header = new Header();
+    }
+
+    @Override
+    public List<T> load(Class<T> clazz) throws IOException {
+        List<T> list = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(getPath(clazz)))) {
+            String csvT;
+            int i = 0;
+            while((csvT = bufferedReader.readLine()) != null) {
+                if (i != 0) {
+                    list.add(convertStringToEntity(clazz, csvT));
+                }
+                i++;
+            }
+        }
+        return list;
+    }
+
+    @Override
+    public void save(List<T> list) throws IOException {
+        Class<?> clazz = list.get(0).getClass();
+        CSVWriter writer = new CSVWriter(new FileWriter(getPath(clazz)),
+                ';',
+                '\'',
+                '\\',
+                "\n");
+        writer.writeNext(getHeader(clazz));
+        for (T t : list) {
+            if (t != null) {
+                String[] tStrings = convertEntityToString(t);
+                writer.writeNext(tStrings);
+            }
+        }
+        System.out.println(clazz.getName() + DATA_SAVED);
+        writer.close();
+    }
+
+    private String getPath(Class<?> clazz) {
+        if (clazz.equals(Room.class) || clazz.equals(HotelFacility.class)) {
```

Comment:
it's not a good way to implement logic, my expectations were a little bit different. Better to leave in ReaderWriter logic to writing/reading a file and pass converter interface inside that will contain all methods to convert and write for specific object 

public class Converter<T> {

String getFilename();

String[] convertToString(T convertable);

T convertToObject(String[] fields);

}

Reply:
I agree with you, Mark. 
You suggestion much better for OOP... In my implementation the Single responsibility principle doesn't work in ReaderWriter: too many actions for one class.
I could implement your comment if necessery.

Reply:
yes please, just for practice, you can not commit it if you want